src/Plugin/Validation/Constraint/ValidOgMembershipReferenceConstraintValidator.php

```diff
@@ -20,21 +20,31 @@ public function validate($value, Constraint $constraint) {
       return;
     }
 
-    $entity = \Drupal::entityTypeManager()
+    $group = \Drupal::entityTypeManager()
       ->getStorage($value->getFieldDefinition()->getFieldStorageDefinition()->getSetting('target_type'))
       ->load($value->get('target_id')->getValue());
 
-    if (!$entity) {
-      // Entity with that entity ID does not exists. This could happen if a
-      // stale entity is passed for validation.
+    if (!$group) {
+      // Entity with that group ID does not exists. This could happen if a
+      // stale group is passed for validation.
       return;
     }
 
-    $params['%label'] = $entity->label();
+    $params['%label'] = $group->label();
 
-    if (!Og::isGroup($entity->getEntityTypeId(), $entity->bundle())) {
+    if (!Og::isGroup($group->getEntityTypeId(), $group->bundle())) {
       $this->context->addViolation($constraint->NotValidGroup, $params);
     }
+
+    /** @var \Drupal\Core\Entity\ContentEntityInterface $entity */
+    $entity = $this->context->getRoot()->getValue();
+
+    /** @var \Drupal\Core\Access\AccessResult $access */
+    $access = \Drupal::service('og.access')->userAccessEntity('create ' . $entity->getEntityTypeId() . ' ' . $entity->bundle(), $group, \Drupal::currentUser()->getAccount());
```

Comment:
This changes the permission format, but does not update https://github.com/Gizra/og/blob/8.x-1.x/src/EventSubscriber/OgEventSubscriber.php#L175 which defines it as `create $bundle_id content` for nodes. 
So the check is always false because `"create $bundle_id content" != "create node $bundle_id"`

Reply:
That's another big change I'd like to sovle with a follow up issue.

Reply:
@RoySegall, any reason not to restore this in this PR?

```suggestion
    $permission = "create $entity->bundle() content";
    $user = \Drupal::currentUser()->getAccount();
    $access = \Drupal::service('og.access')->userAccessEntity($permission, $group, $user);
```